### PR TITLE
New module: azure_rm_ddosprotectionplan

### LINKED ISSUE #492: Add support for DDoS Protection Plan
**SUMMARY**
we would also like to create our DDoS protection plan with ansible. It seems currently there is no module for that.
https://docs.microsoft.com/en-us/azure/ddos-protection/ddos-protection-overview

**ISSUE TYPE**
- Feature Idea

**COMPONENT NAME**
None as of now, most likely something like azure_rm_ddos_protection_plan

### PR DESCRIPTION
**Pull request by ikarldasan**
##### SUMMARY
This module introduces a wide variety of configurations for creating a DDoS protection plan on Azure.
Create DDoS protection plan on Azure.
Update DDoS protection plan on Azure.
Delete DDoS protection plan on Azure.
Get DDoS protection plan on Azure.
List DDoS protection plan in Azure Resource Group.
List DDoS protection plan in Azure Subscription.

Fixes #492 

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
azure_rm_ddos_protection_plan

##### ADDITIONAL INFORMATION
This module allows user to create/update or delete the DDoS Protection Plan.
It also provides all the parameters given by:  https://docs.microsoft.com/en-us/azure/ddos-protection/ddos-protection-overview

**More information**
```
azure_rm_ddos_protection_plan:
   name: ddos
   resource_group: testgroupans
   location: eastus2
   tags:
      a: c
```

### CODE REVIEW

**Comment by Fred-sun** [comment]
@ikarldasan Thank for your contribution! Please help to change the module name to azure_rm_ddosprotectionplan*, which is consistent with other modules, thank you!

File: plugins/modules/azure_rm_ddos_protection_plan.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+      description:
+          - Assert the state of the DDoS protection plan.
+          - Use C(present) to create or update an DDoS protection plan and C(absent) to delete it.
+      type: str
+      default: present
+      choices:
+          - absent
+          - present
+extends_documentation_fragment:
+- azure.azcollection.azure
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_ddos_protection_plan.py]:
```suggestion
    - azure.azcollection.azure
    - azure.azcollection.azure_tags
```

File: plugins/modules/azure_rm_ddos_protection_plan.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        type: str
+    state:
+      description:
+          - Assert the state of the DDoS protection plan.
+          - Use C(present) to create or update an DDoS protection plan and C(absent) to delete it.
+      type: str
+      default: present
+      choices:
+          - absent
+          - present
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_ddos_protection_plan.py]: Please add log_path and log_mode parameter to this module! Thank you very much!
```e.g.
    log_path:
        description:
            - parent argument.
        type: str
    log_mode:
        description:
            - parent argument.
        type: str

**Review by Fred-sun — COMMENTED** [review]
@ikarldasan There is no content in " tests/integration/targets/azure_rm_ddos_protection_plan/aliases"! Please share the result of you execute "tests/integration/targets/azure_rm_ddos_protection_plan/tasks/main.yml"!
```e.g.
cloud/azure
shippable/azure/group10
destructive

**Comment by ikarldasan** [comment]
> @ikarldasan There is no content in " tests/integration/targets/azure_rm_ddos_protection_plan/aliases"! Please share the result of you execute "tests/integration/targets/azure_rm_ddos_protection_plan/tasks/main.yml"!
> 
> ```
> cloud/azure
> shippable/azure/group10
> destructive
> ```



@Fred-sun , have addressed the changes suggested.
But facing an issue running the integration tests.
Running this command and getting the following result.

**_ansible-test integration azure_rm_ddosprotectionplan --allow-destructive -e -vvv_**

```
RLIMIT_NOFILE: (1024, 1048576)
MAXFD: -1
Falling back to tests in "tests/integration/targets/" because "roles/test/" was not found.
WARNING: Excluding tests marked "cloud/azure" which require config (see "/home/karl/.local/lib/python3.8/site-packages/ansible_test/config/cloud-config-azure.ini.template"): azure_rm_ddosprotectionplan
WARNING: All targets skipped.
```

Can you please help understand what I may be missing.
Thank you!!

**Comment by ikarldasan** [comment]
@Fred-sun here are the test results, run locally. Running the integration test as playbooks using `ansible-playbook playbook -vvv`. Thank you.

- **Create:**
```
**Playbook**
---
    - hosts: localhost
      connection: local
      tasks:   
       - name: Create DDoS protection plan
         azure.azcollection.azure_rm_ddosprotectionplan:
           client_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           resource_group: rg
           secret: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           subscription_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           tenant: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           location: eastus2
           name: ddos
         register: results

**Result:**

changed: [localhost] => {
    "changed": true,
    "check_mode": false,
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "append_tags": true,
            "auth_source": null,
            "cert_validation_mode": null,
            "client_id": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "cloud_environment": "AzureCloud",
            "location": "eastus2",
            "log_mode": null,
            "log_path": null,
            "name": "ddos",
            "password": null,
            "profile": null,
            "resource_group": "rg",
            "secret": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "state": "present",
            "subscription_id": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
            "tags": null,
            "tenant": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
        }
    },
    "state": {
        "additional_properties": null,
        "etag": "W/\"\"",
        "id": "/subscriptions/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/resourceGroups/rg/providers/Microsoft.Network/ddosProtectionPlans/ddos",
        "location": "eastus2",
        "name": "ddos",
        "provisioning_state": "Succeeded",
        "resource_guid": null,
        "tags": null,
        "type": "Microsoft.Network/ddosProtectionPlans",
        "virtual_networks": null
    }
}
META: ran handlers
META: ran handlers

PLAY RECAP *************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   


```

-  **Create (check mode):**
```
**Playbook:**
---
    - hosts: localhost
      connection: local
      tasks:   
       - name: Create DDoS protection plan (check mode)
         azure.azcollection.azure_rm_ddosprotectionplan:
           client_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           resource_group: rg
           secret: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           subscription_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           tenant: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           location: eastus2
           name: ddos
         register: results
         check_mode: yes


**Result:**

ok: [localhost] => {
    "changed": false,
    "check_mode": true,
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "append_tags": true,
            "auth_source": null,
            "cert_validation_mode": null,
            "client_id": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "cloud_environment": "AzureCloud",
            "location": "eastus2",
            "log_mode": null,
            "log_path": null,
            "name": "ddos",
            "password": null,
            "profile": null,
            "resource_group": "rg",
            "secret": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "state": "present",
            "subscription_id": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
            "tags": null,
            "tenant": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
        }
    },
    "state": {
        "additional_properties": null,
        "etag": "W/\"\"",
        "id": "/subscriptions/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/resourceGroups/rg/providers/Microsoft.Network/ddosProtectionPlans/ddos",
        "location": "eastus2",
        "name": "ddos",
        "provisioning_state": "Succeeded",
        "resource_guid": null,
        "tags": {
            "a": "c"
        },
        "type": "Microsoft.Network/ddosProtectionPlans",
        "virtual_networks": null
    }
}
META: ran handlers
META: ran handlers

PLAY RECAP *************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

```
-  **Create (Idempotent):**
```
**Playbook**

---
    - hosts: localhost
      connection: local
      tasks:   
       - name: Create DDoS protection plan(Idempotent)
         azure.azcollection.azure_rm_ddosprotectionplan:
           client_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           resource_group: rg
           secret: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           subscription_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           tenant: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           location: eastus2
           name: ddos
         register: results

**Result:**
ok: [localhost] => {
    "changed": false,
    "check_mode": false,
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "append_tags": true,
            "auth_source": null,
            "cert_validation_mode": null,
            "client_id": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "cloud_environment": "AzureCloud",
            "location": "eastus2",
            "log_mode": null,
            "log_path": null,
            "name": "ddos",
            "password": null,
            "profile": null,
            "resource_group": "rg",
            "secret": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "state": "present",
            "subscription_id": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
            "tags": null,
            "tenant": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
        }
    },
    "state": {
        "additional_properties": null,
        "etag": "W/\"\"",
        "id": "/subscriptions/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/resourceGroups/rg/providers/Microsoft.Network/ddosProtectionPlans/ddos",
        "location": "eastus2",
        "name": "ddos",
        "provisioning_state": "Succeeded",
        "resource_guid": null,
        "tags": {},
        "type": "Microsoft.Network/ddosProtectionPlans",
        "virtual_networks": null
    }
}
META: ran handlers
META: ran handlers

PLAY RECAP *************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

```

- **Update:**

```
**Playbook:**
---
    - hosts: localhost
      connection: local
      tasks:   
       - name: Updates DDoS protection plan 
         azure.azcollection.azure_rm_ddosprotectionplan:
           client_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           resource_group: rg
           secret: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           subscription_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           tenant: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           location: eastus2
           name: ddos
           tags:
            a: c
         register: results


**Result:**

changed: [localhost] => {
    "changed": true,
    "check_mode": false,
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "append_tags": true,
            "auth_source": null,
            "cert_validation_mode": null,
            "client_id": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "cloud_environment": "AzureCloud",
            "location": "eastus2",
            "log_mode": null,
            "log_path": null,
            "name": "ddos",
            "password": null,
            "profile": null,
            "resource_group": "rg",
            "secret": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "state": "present",
            "subscription_id": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
            "tags": {
                "a": "c"
            },
            "tenant": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
        }
    },
    "state": {
        "additional_properties": null,
        "etag": "W/\"\"",
        "id": "/subscriptions/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/resourceGroups/rg/providers/Microsoft.Network/ddosProtectionPlans/ddos",
        "location": "eastus2",
        "name": "ddos",
        "provisioning_state": "Succeeded",
        "resource_guid": null,
        "tags": {
            "a": "c"
        },
        "type": "Microsoft.Network/ddosProtectionPlans",
        "virtual_networks": null
    }
}
META: ran handlers
META: ran handlers

PLAY RECAP *************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

```

- **Delete (Check_Mode):**
```
**Playbook**
---
    - hosts: localhost
      connection: local
      tasks:
       - name: Delete DDoS Protection Plan (Check Mode)
         azure.azcollection.azure_rm_ddosprotectionplan:
           client_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           secret: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           subscription_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           tenant: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           resource_group: rg
           name:  ddos
           state: absent
           location: eastus2
         check_mode: yes


**Result:**
changed: [localhost] => {
    "changed": true,
    "check_mode": true,
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "append_tags": true,
            "auth_source": null,
            "cert_validation_mode": null,
            "client_id": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "cloud_environment": "AzureCloud",
            "location": "eastus2",
            "log_mode": null,
            "log_path": null,
            "name": "ddos",
            "password": null,
            "profile": null,
            "resource_group": "rg",
            "secret": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "state": "absent",
            "subscription_id": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
            "tags": null,
            "tenant": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
        }
    },
    "state": {
        "additional_properties": null,
        "etag": "W/\"\"",
        "id": "/subscriptions/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/resourceGroups/rg/providers/Microsoft.Network/ddosProtectionPlans/ddos",
        "location": "eastus2",
        "name": "ddos",
        "provisioning_state": "Succeeded",
        "resource_guid": null,
        "tags": null,
        "type": "Microsoft.Network/ddosProtectionPlans",
        "virtual_networks": null
    }
}
META: ran handlers
META: ran handlers

PLAY RECAP *************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0  

```

- **Delete**
```
**Playbook**
---
    - hosts: localhost
      connection: local
      tasks:
       - name: Delete DDoS Protection Plan
         azure.azcollection.azure_rm_ddosprotectionplan:
           client_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           secret: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           subscription_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           tenant: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           resource_group: rg
           name:  ddos
           state: absent
           location: eastus2

**Result:**
changed: [localhost] => {
    "changed": true,
    "check_mode": false,
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "append_tags": true,
            "auth_source": null,
            "cert_validation_mode": null,
            "client_id": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "cloud_environment": "AzureCloud",
            "location": "eastus2",
            "log_mode": null,
            "log_path": null,
            "name": "ddos",
            "password": null,
            "profile": null,
            "resource_group": "rg",
            "secret": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "state": "absent",
            "subscription_id": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
            "tags": null,
            "tenant": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
        }
    },
    "state": {
        "additional_properties": null,
        "etag": "W/\"\"",
        "id": "/subscriptions/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/resourceGroups/rg/providers/Microsoft.Network/ddosProtectionPlans/ddos",
        "location": "eastus2",
        "name": "ddos",
        "provisioning_state": "Succeeded",
        "resource_guid": null,
        "status": "Deleted",
        "tags": null,
        "type": "Microsoft.Network/ddosProtectionPlans",
        "virtual_networks": null
    }
}
META: ran handlers
META: ran handlers

PLAY RECAP *************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0 

```

-  **Get by name:**
```
**Playbook**
---
    - hosts: localhost
      connection: local
      tasks:
       - name: Retrieve DDoS Protection Plan
         azure.azcollection.azure_rm_ddosprotectionplan_info:
           resource_group: rg
           client_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           resource_group: rg
           secret: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           subscription_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           tenant: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           name: ddos
         register: results

**Result:**
ok: [localhost] => {
    "changed": false,
    "ddosprotectionplan": [
        {
            "additional_properties": null,
            "etag": "W/\"\"",
            "id": "/subscriptions/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/resourceGroups/rg/providers/Microsoft.Network/ddosProtectionPlans/ddos",
            "location": "eastus2",
            "name": "ddos",
            "provisioning_state": "Succeeded",
            "resource_guid": null,
            "tags": {
                "a": "c"
            },
            "type": "Microsoft.Network/ddosProtectionPlans",
            "virtual_networks": null
        }
    ],
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "auth_source": null,
            "cert_validation_mode": null,
            "client_id": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "cloud_environment": "AzureCloud",
            "name": "ddos",
            "password": null,
            "profile": null,
            "resource_group": "rg",
            "secret": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "subscription_id": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
            "tenant": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
        }
    }
}
META: ran handlers
META: ran handlers

PLAY RECAP *************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

```

- **Get by resource group:**
```
**Playbook:**
---
    - hosts: localhost
      connection: local
      tasks:
       - name: Retrieve DDoS Protection Plan (By Resource Group)
         azure.azcollection.azure_rm_ddosprotectionplan_info:
           resource_group: rg
           client_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           resource_group: rg
           secret: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           subscription_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           tenant: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
         register: results

**Result:**
ok: [localhost] => {
    "changed": false,
    "ddosprotectionplan": [
        {
            "additional_properties": null,
            "etag": "W/\"\"",
            "id": "/subscriptions/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/resourceGroups/rg/providers/Microsoft.Network/ddosProtectionPlans/ddos",
            "location": "eastus2",
            "name": "ddos",
            "provisioning_state": "Succeeded",
            "resource_guid": null,
            "tags": {
                "a": "c"
            },
            "type": "Microsoft.Network/ddosProtectionPlans",
            "virtual_networks": null
        }
    ],
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "auth_source": null,
            "cert_validation_mode": null,
            "client_id": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "cloud_environment": "AzureCloud",
            "name": null,
            "password": null,
            "profile": null,
            "resource_group": "rg",
            "secret": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "subscription_id": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
            "tenant": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
        }
    }
}
META: ran handlers
META: ran handlers

PLAY RECAP *************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

```

- **Get by subscription:**
```
**Playbook:**
---
    - hosts: localhost
      connection: local
      tasks:
       - name: Retrieve DDoS Protection Plan (By Subscription)
         azure.azcollection.azure_rm_ddosprotectionplan_info:
           client_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           secret: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           subscription_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
           tenant: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
         register: results

**Result:**
ok: [localhost] => {
    "changed": false,
    "ddosprotectionplan": [
        {
            "additional_properties": null,
            "etag": "W/\"\"",
            "id": "/subscriptions/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/resourceGroups/rg/providers/Microsoft.Network/ddosProtectionPlans/ddos",
            "location": "eastus2",
            "name": "ddos",
            "provisioning_state": "Succeeded",
            "resource_guid": null,
            "tags": {
                "a": "c"
            },
            "type": "Microsoft.Network/ddosProtectionPlans",
            "virtual_networks": null
        }
    ],
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "auth_source": null,
            "cert_validation_mode": null,
            "client_id": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "cloud_environment": "AzureCloud",
            "name": null,
            "password": null,
            "profile": null,
            "resource_group": null,
            "secret": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "subscription_id": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
            "tenant": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
        }
    }
}
META: ran handlers
META: ran handlers

PLAY RECAP *************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   




```

**Comment by Fred-sun** [comment]
@ikarldasan Please add following line to ignore2.11.txt, 2.9.txt, 2.10.txt and 2.12.txt. Thank you very much!
```e.g.
plugins/modules/azure_rm_ddosprotectionplan_info.py validate-modules:required_if-requirements-unknown
plugins/modules/azure_rm_ddosprotectionplan_info.py validate-modules:required_if-unknown-key

**Comment by ikarldasan** [comment]
@Fred-sun have addressed the changes suggested.
Thank you

**Comment by Fred-sun** [comment]
> @Fred-sun have addressed the changes suggested.
> Thank you

Thanks for your udpate! I will recheck it!

**Review by Ryujinx01 — DISMISSED** [review]

**Comment by Fred-sun** [comment]
@ikarldasan Please help to delete all ignores on ignore2.11.txt, ignore2.9.txt, ignore2.10.txt and ignore2.12.txt that you add. It has been fixed by #540!

File: plugins/modules/azure_rm_ddosprotectionplan.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+            log_mode=dict(type='str')
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+        self.state = None
+        self.tags = None
+        self.log_path = None
+        self.log_mode = None
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_ddosprotectionplan.py]:
```suggestion

```

File: plugins/modules/azure_rm_ddosprotectionplan.py [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+    def __init__(self):
+        # define user inputs from playbook
+
+        self.module_arg_spec = dict(
+            resource_group=dict(type='str', required=True),
+            name=dict(type='str', required=True),
+            location=dict(type='str'),
+            state=dict(choices=['present', 'absent'],
+                       default='present', type='str'),
+            log_path=dict(type='str'),
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_ddosprotectionplan.py]:
```suggestion

```

File: plugins/modules/azure_rm_ddosprotectionplan.py [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+            log_mode=dict(type='str')
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_ddosprotectionplan.py]: Please delete paramter log_path and log_mode, It has been add by #540 ( praent class)

```suggestion

```

File: tests/integration/targets/azure_rm_ddosprotectionplan/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,83 @@
+- name: Create random ddos protection plan
+  set_fact:
+      ddosprotectionplan: "test{{ resource_group | hash('md5') | truncate(16, True, '') + (65535 | random | string) }}"
+
+- name: Create DDoS protection plan (check mode)
+  azure_rm_ddosprotectionplan:
+    location: eastus2
+    name: "{{ ddosprotectionplan }}"
+    resource_group: "{{ resource_group }}"
+    check_mode: yes
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_ddosprotectionplan/tasks/main.yml]:
```suggestion
  check_mode: yes
```

File: tests/integration/targets/azure_rm_ddosprotectionplan/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  register: results
+
+- assert:
+    that: 
+      - results.changed
+      - results.state.tags.test == 'modified'
+
+- name: Retrieve DDoS protection plan
+  azure_rm_ddosprotectionplan_info:
+    location: eastus2
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_ddosprotectionplan/tasks/main.yml]: azure_rm_ddosprotectionplan_info.py not support paramter 'location'!
```suggestion
```

File: tests/integration/targets/azure_rm_ddosprotectionplan/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    resource_group: "{{ resource_group }}"
+  register: results
+
+- name: Assert that facts module returned result
+  assert:
+    that: 
+      - results.ddosprotectionplan[0].tags.test == 'modified'
+
+- name: Test idempotent
+  azure_rm_ddosprotectionplan_info:
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_ddosprotectionplan/tasks/main.yml]:
```suggestion
  azure_rm_ddosprotectionplan:
```

**Comment by ikarldasan** [comment]
@Fred-sun have addressed the changes suggested.
Thank you

**Comment by Fred-sun** [comment]
@ikarldasan Please help to delete all ignores on ignore2.11.txt, ignore2.9.txt, ignore2.10.txt and ignore2.12.txt that you add. It has been fixed by #540!

File: plugins/modules/azure_rm_ddosprotectionplan.py [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+#
+# Copyright (c) 2021 Praveen Ghuge (@praveenghuge), Karl Dasan (@ikarldasan)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+DOCUMENTATION = '''
+---
+module: azure_rm_ddosprotectionplan
+version_added: "0.1.2"
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_ddosprotectionplan.py]:
```suggestion
version_added: "1.7.0"
```

File: plugins/modules/azure_rm_ddosprotectionplan_info.py [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_ddosprotectionplan_info
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_ddosprotectionplan_info.py:14]:
```suggestion
module: azure_rm_ddosprotectionplan_info
version_added: "1.7.0"
```

**Review by Fred-sun — COMMENTED** [review]
@ikarldasan One last update requirement, Thank you very much!

**Comment by ikarldasan** [comment]
> @ikarldasan One last update requirement, Thank you very much!

@Fred-sun Have updated the changes suggested. Thank you

**Comment by ikarldasan** [comment]
> @ikarldasan One last update requirement, Thank you very much!

@Fred-sun Do we need any more enhancements? Can we merge this?  Thank you

**Comment by Fred-sun** [comment]
> > @ikarldasan One last update requirement, Thank you very much!
> 
> @Fred-sun Do we need any more enhancements? Can we merge this? Thank you

@ikarldasan I will move forward with the merger as soon as possible. Thank you very much!

**Pull request merged**